src/Particular.ThroughputCollector/ThroughputCollectorHostBuilderExtensions.cs

```diff
@@ -1,86 +1,97 @@
-                    break;
-                case "SQLServer":
-                    broker = Contracts.Broker.SqlServer;
-                    break;
-                case "AmazonSQS":
-                    broker = Contracts.Broker.AmazonSQS;
-                    break;
-            }
+        services.AddSingleton(new ThroughputSettings(broker: broker, transportConnectionString: transportConnectionString, serviceControlAPI: serviceControlAPI, serviceControlQueue: serviceControlQueue, errorQueue: errorQueue, persistenceType: persistenceType, customerName: customerName, serviceControlVersion: serviceControlVersion, auditQueue: auditQueue));
+        services.AddHostedService<AuditThroughputCollectorHostedService>();
+        services.AddSingleton<IThroughputCollector, ThroughputCollector>();
+        services.AddSingleton<ThroughputController>();
+        switch (broker)
+        {
+            case Contracts.Broker.AmazonSQS:
+                services.AddSingleton<IThroughputQuery, AmazonSQSQuery>();
+                services.AddHostedService<BrokerThroughputCollectorHostedService>();
+                break;
+            case Contracts.Broker.RabbitMQ:
+                services.AddSingleton<IThroughputQuery, RabbitMQQuery>();
+                services.AddHostedService<BrokerThroughputCollectorHostedService>();
+                break;
+            case Contracts.Broker.AzureServiceBus:
+                services.AddSingleton<IThroughputQuery, AzureQuery>();
+                services.AddHostedService<BrokerThroughputCollectorHostedService>();
+                break;
+            case Contracts.Broker.SqlServer:
+                services.AddSingleton<IThroughputQuery, SqlServerQuery>();
+                services.AddHostedService<BrokerThroughputCollectorHostedService>();
+                break;
+        }
 
-            //try to ensure the serviceControlAPI is correct (https://learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints?view=aspnetcore-8.0#url-formats)
-            if (serviceControlAPI.StartsWith("http://*"))
-            {
-                serviceControlAPI = serviceControlAPI.Replace("http://*", "http://localhost");
-            }
-            else if (serviceControlAPI.StartsWith("http://0.0.0.0"))
+        hostBuilder.AddThroughputCollectorPersistence(persistenceType, errorQueue, serviceControlQueue);
+
+        return hostBuilder;
+    }
+
+    public static IHostApplicationBuilder AddThroughputCollectorPersistence(this IHostApplicationBuilder hostBuilder, string persistenceType, string? errorQueue = null, string? serviceControlQueue = null)
+    {
+        hostBuilder.Services.AddPersistence(persistenceType, settings =>
+        {
+            if (!string.IsNullOrEmpty(errorQueue))
             {
-                serviceControlAPI = serviceControlAPI.Replace("http://0.0.0.0", "http://localhost");
+                settings.PlatformEndpointNames.Add(errorQueue);
```

Comment:
Why are we storing the platform queue names twice? on here as well as on the throughputSettings class.

Reply:
I need the queue names in the persistence assemblies to do DB side filtering of the queues, and `ThroughputSettings` is not available.

I left them on ThroughputSettings because `BrokerThroughputCollectorHostedService` needs them as well.

Reply:
I'm thinking perhaps we don't save those queues at ? I don't think that @johnsimons is doing that on the broker side. That would mean we wouldn't need these on the persistence side.

Reply:
I'm actually thinking the opposite in that they should be included in the summary as well, just marked as a platform queue (for lack of a better term).  Although how we treat platform queues is probably better discussed as a separate topic outside this PR.

Reply:
It seems we need to move `ThroughputSettings` to `Particular.ThroughputCollector.Contracts`

Reply:
@abparticular that has been moved. The existing tool removes those error etc queues and does not include them in the report (for most transports - it's not consistent).